Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift

```diff
@@ -169,23 +167,44 @@ final class NetworkProtectionConnectionTester {
 
     // MARK: - Timer scheduling
 
-    private func scheduleTimer() async {
+    private func scheduleTimer(testImmediately: Bool) async throws {
         await stopScheduledTimer()
 
+        await timerRunCoordinator.start()
+
+        if testImmediately {
+            do {
+                try await testConnection(isStartupTest: true)
+            } catch {
+                os_log("Rethrowing exception", log: log)
+                throw error
+            }
+        }
```

Comment:
We test the connection before setting the `DispatchTimer` up, as the timer doesn't like being created but not run, and crashes.